core/src/main/java/io/fabric8/maven/core/service/PortForwardService.java

```diff
@@ -0,0 +1,256 @@
+                    List<Pod> candidatePods;
+                    if (nextForwardedPod[0] != null) {
+                        candidatePods = new LinkedList<>();
+                        candidatePods.add(nextForwardedPod[0]);
+                        candidatePods.add(pod);
+                    } else {
+                        candidatePods = Collections.singletonList(pod);
+                    }
+                    Pod newPod = getNewestPod(candidatePods); // may be null
+                    if (!podEquals(nextForwardedPod[0], newPod)) {
+                        nextForwardedPod[0] = newPod;
+                        podChanged.signal();
+                    }
+                } finally {
+                    monitor.unlock();
+                }
+            }
+
+            @Override
+            public void onClose(KubernetesClientException e) {
+                // don't care
+            }
+        });
+
+        forwarderThread.start();
+
+        final Closeable handle = new Closeable() {
+            @Override
+            public void close() throws IOException {
+                try {
+                    watch.close();
+                } catch (Exception e) {}
+                try {
+                    forwarderThread.interrupt();
+                    forwarderThread.join(15000);
+                } catch (Exception e) {}
+            }
+        };
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run() {
+                try {
+                    handle.close();
+                } catch (Exception e) {
+                    // suppress
+                }
+            }
+        });
+
+        return handle;
+    }
+
+    private boolean podEquals(Pod pod1, Pod pod2) {
+        if (pod1 == pod2) {
+            return true;
+        }
+        if (pod1 == null || pod2 == null) {
+            return false;
+        }
+        return KubernetesHelper.getName(pod1).equals(KubernetesHelper.getName(pod2));
+    }
+
+    private Pod getNewestPod(LabelSelector selector) {
+        FilterWatchListDeletable<Pod, PodList, Boolean, Watch, Watcher<Pod>> pods =
+                KubernetesClientUtil.withSelector(kubernetes.pods(), selector, log);
+
+        PodList list = pods.list();
+        if (list != null) {
+            List<Pod> items = list.getItems();
+            return getNewestPod(items);
+        }
+        return null;
+    }
+
+    private Pod getNewestPod(List<Pod> items) {
+        Pod targetPod = null;
+        if (items != null) {
+            for (Pod pod : items) {
+                PodStatusType status = KubernetesHelper.getPodStatus(pod);
+                switch (status) {
+                case WAIT:
+                case OK:
+                    if (targetPod == null || (KubernetesHelper.isPodReady(pod) && KubernetesResourceUtil.isNewerResource(pod, targetPod))) {
+                        targetPod = pod;
+                    }
+                    break;
```

Comment:
It's the break of the switch... they should be equivalent, btw.. better changing this.. :)